cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo.go

```diff
@@ -562,17 +562,34 @@ func (s *Server) deleteRepo(ctx context.Context, repoRef *corev1.PackageReposito
 
 	log.V(4).Infof("Deleting repo: [%s]", repoRef.Identifier)
 
+	// delete secrets, if applicable, for kubeapps-managed secrets environment
+	var deleteSecret string
+	if !s.pluginConfig.UserManagedSecrets {
+		key := types.NamespacedName{Namespace: repoRef.Context.Namespace, Name: repoRef.Identifier}
+		var repo sourcev1.HelmRepository
+		if err = client.Get(ctx, key, &repo); err != nil {
+			return statuserror.FromK8sError("get", "HelmRepository", repoRef.Identifier, err)
+		}
+		if repo.Spec.SecretRef != nil && repo.Spec.SecretRef.Name != "" {
+			deleteSecret = repo.Spec.SecretRef.Name
+		}
+	}
+
 	repo := &sourcev1.HelmRepository{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      repoRef.Identifier,
 			Namespace: repoRef.Context.Namespace,
 		},
 	}
-
-	// TODO (gfichtenholt) secrets, if applicable for kubeapps-managed secrets environment
-
 	if err = client.Delete(ctx, repo); err != nil {
 		return statuserror.FromK8sError("delete", "HelmRepository", repoRef.Identifier, err)
+	} else if deleteSecret != "" {
+		typedClient, err := s.clientGetter.Typed(ctx, s.kubeappsCluster)
+		if err != nil {
+			return err
+		} else if err = typedClient.CoreV1().Secrets(repoRef.Context.Namespace).Delete(ctx, deleteSecret, metav1.DeleteOptions{}); err != nil {
+			return statuserror.FromK8sError("delete", "secret", deleteSecret, err)
+		}
```

Comment:
let me look into this owner references business. It's new to me.

Reply:
yeah, it works. I implemented ownerReferences as you suggested. It was a little tricky because:
1) I need to create a secret so that I can create a repo that references it
2) create a repo 
3) update the secret to set the owner reference to (2). 
I had to do it in that order because to set the owner reference you need the object (i.e. repo) UID which I don't have until (2) is complete.

Reply:
Excellent, nice one Greg. I think that'll be neater, having K8s garbage collect, and more importantly, we won't be deleting secrets created by users outside of Kubeapps. Thanks.

Reply:
👍